File: 1833-find-the-highest-altitude/solution.java
class Solution {
    public int largestAltitude(int[] gain) {
        int n = gain.length;
        int[] prefixGain = new int[n+1];
        prefixGain[0] = 0;
        int maxGain = 0;

        for(int i=1; i<=n; i++) {
            prefixGain[i] = prefixGain[i-1] + gain[i-1];//-5
            maxGain = Math.max(maxGain, prefixGain[i]);//-1
        }
        return maxGain;
    }
}

File: 1833-find-the-highest-altitude/solution.swift
class Solution {
    func largestAltitude(_ gain: [Int]) -> Int {
        var n = gain.count
        var ans = Array(repeating: 0, count: n+1)
        var maxAns = 0
        ans[0] = 0

        for i in 1...n {
            ans[i] = ans[i-1] + gain[i-1]
            maxAns = max(maxAns, ans[i])
        }
        return maxAns
    }
}
